packages/notebook/src/browser/notebook-editor-widget.tsx

```diff
@@ -165,4 +174,23 @@ export class NotebookEditorWidget extends ReactWidget implements Navigatable, Sa
         super.onAfterDetach(msg);
         this.notebookEditorService.removeNotebookEditor(this);
     }
+
+    postKernelMessage(message: unknown): void {
+        this.onPostKernelMessageEmitter.fire(message);
+    }
+
+    postRendererMessage(rendererId: string, message: unknown): void {
+        this.onPostRendererMessageEmitter.fire({ rendererId, message });
+    }
+
+    recieveKernelMessage(message: unknown): void {
+        this.onDidRecieveKernelMessageEmitter.fire(message);
+    }
+
+    override dispose(): void {
+        this.onDidChangeModelEmitter.dispose();
+        this.onPostKernelMessageEmitter.dispose();
```

Comment:
Minor: Missing emitter disposal?

```suggestion
        this.onPostKernelMessageEmitter.dispose();
        this.onPostRendererMessageEmitter.dispose();
```